packages/engine/src/scene/systems/ShadowSystem.tsx

```diff
@@ -451,6 +434,41 @@ const execute = () => {
     return
   }
 
+  for (const entity of directionalLightQuery.enter()) {
+    const uuid = getComponent(entity, UUIDComponent)
+    directionalLightMap.set(entity, uuid)
+  }
+
+  for (const entity of directionalLightQuery.exit()) {
+    const uuid = directionalLightMap.get(entity)
+    if (!uuid) continue
+
+    // CSM directional lights should be their own component
+    for (const renderSettingsEntity of renderSettingsQuery()) {
+      const renderSettings = getComponent(renderSettingsEntity, RenderSettingsComponent)
+      if (renderSettings.primaryLight === uuid) {
+        getMutableComponent(renderSettingsEntity, RenderSettingsComponent).merge({
+          csm: false,
+          primaryLight: '' as EntityUUID
+        })
+        if (getState(EngineState).isEditing) {
+          const source = getComponent(renderSettingsEntity, SourceComponent)
```

Comment:
Let's move this logic to the studio in a reactor, since it really is an authoring layer concept.